owncloudApp/src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.kt

```diff
@@ -199,227 +178,207 @@ class FileDataStorageManager {
-            } finally {
-                cursorOnShared?.close()
-            }
-
-            if (allSharedFiles.isNotEmpty()) {
-                val allSharedDirs = Vector<Long>()
-                for (file in allSharedFiles) {
-                    if (file.isFolder) {
-                        allSharedDirs.add(file.fileId)
-                    }
-                }
-                for (file in allSharedFiles) {
-                    if (file.isFolder || (!file.isFolder && !allSharedDirs.contains(file.parentId))) {
-                        result.add(file)
-                    }
-                }
-            }
-            result.sort()
-            return result
+            return Vector()
+            // FIXME: 13/10/2020 : New_arch: Shared By Link
+//            val allSharedFiles = Vector<OCFile>()
+//            val result = Vector<OCFile>()
+//            var cursorOnShared: Cursor? = null
+//            try {
+//                cursorOnShared = contentResolver?.query(
+//                    CONTENT_URI,
+//                    null,
+//                    "$FILE_SHARED_VIA_LINK = ? AND $FILE_ACCOUNT_OWNER = ? ",
+//                    arrayOf(1.toString(), account.name),
+//                    null
+//                )
+//                if (cursorOnShared != null && cursorOnShared.moveToFirst()) {
+//                    var file: OCFile?
+//                    do {
+//                        file = createFileInstance(cursorOnShared)
+//                        allSharedFiles.add(file)
+//                    } while (cursorOnShared.moveToNext())
+//                }
+//            } catch (exception: Exception) {
+//                Timber.e(exception, "Exception retrieving all the shared by link files")
+//            } finally {
+//                cursorOnShared?.close()
+//            }
+//
+//            if (allSharedFiles.isNotEmpty()) {
+//                val allSharedDirs = Vector<Long>()
+//                for (file in allSharedFiles) {
+//                    if (file.isFolder) {
+//                        allSharedDirs.add(file.fileId)
+//                    }
+//                }
+//                for (file in allSharedFiles) {
+//                    if (file.isFolder || (!file.isFolder && !allSharedDirs.contains(file.parentId))) {
+//                        result.add(file)
+//                    }
+//                }
+//            }
+//            result.sort()
+//            return result
         }
 
-    fun getFileByPath(path: String): OCFile? {
-        val c = getFileCursorForValue(FILE_PATH, path)
-        var file: OCFile? = null
-        if (c != null) {
-            if (c.moveToFirst()) {
-                file = createFileInstance(c)
-            }
-            c.close()
-        }
-        return if (file == null && ROOT_PATH == path) {
-            createRootDir() // root should always exist
-        } else file
-    }
+    fun getFileByPath(path: String): OCFile? = runBlocking {
```

Comment:
Have you tried to solve this scenario without using `runBlocking`? I know the oC app is a special case because it has Java and Kotlin classes and we might need to call from Java to Kotlin. It's normal to use `runBlocking` in testing but not in the rest of the code because it blocks the thread (UI or not), so we lose the async approach...

Reply:
The idea is to remove FileDataStorageManager and call those usecases from FilesViewModel. I have added some todos, and added IO as runblocking context. This is an intermediate step, it won't be that way in the final version.